File: backend/services/transcribe-summarize-service/startup.ts
import { verifyServices } from "../../shared/preflight";
import { startMQ } from "../../shared/message-queue/messageQueue";
import { pingDb } from "../../shared/db";
import { pingBucket } from "../../shared/bucket";
import { pingAi } from "../../shared/ai/ai_client";

/**
 * Fail-fast preflight for the worker: aborts startup if any third-party
 * dependency is unavailable.
 */
export function verifyWorkerServices(): Promise<void> {
  return verifyServices([
    { name: "RabbitMQ", check: startMQ },
    { name: "Postgres", check: pingDb },
    { name: "Supabase Storage", check: pingBucket },
    { name: "OpenRouter", check: pingAi },
  ]);
}

File: youtube-fetcher/app/startup.py
"""Fail-fast preflight for the fetcher, mirroring
backend/services/api/startup.ts: aborts startup if any dependency is
unavailable."""

import bucket
from contract import contract
from message_queue import mq
from preflight import verify_services


def verify_fetcher_services() -> None:
    # The contract comes first on its own: the queue names and bucket name
    # both come from it, so the other checks can't run without it.
    verify_services([("API /contract", contract.load)])
    verify_services(
        [
            ("RabbitMQ", mq.connect),
            ("Supabase Storage", bucket.ping_bucket),
        ]
    )
